File: backend/internal/domain/models/notification.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type NotificationType string

const (
	NotificationTypeVisitReminder NotificationType = "visit_reminder"
	NotificationTypeFollowUp      NotificationType = "follow_up"
	NotificationTypeDealUpdate     NotificationType = "deal_update"
	NotificationTypeApproval      NotificationType = "approval"
	NotificationTypeBroadcast     NotificationType = "broadcast"
	NotificationTypeSystem        NotificationType = "system"
)

type Notification struct {
	ID         uuid.UUID        `json:"id"`
	UserID     uuid.UUID        `json:"user_id"`
	Type       NotificationType `json:"type"`
	Title      string           `json:"title"`
	Body       string           `json:"body"`
	EntityType string           `json:"entity_type"`
	EntityID   *uuid.UUID       `json:"entity_id"`
	IsRead     bool             `json:"is_read"`
	ReadAt     *time.Time       `json:"read_at"`
	SentAt     *time.Time       `json:"sent_at"`
	CreatedAt  time.Time        `json:"created_at"`
}

File: mobile/lib/features/notifications/domain/entities/notification.dart
import 'package:equatable/equatable.dart';

class AppNotification extends Equatable {
  final String id;
  final String type;
  final String title;
  final String body;
  final String? entityType;
  final String? entityId;
  final bool isRead;
  final DateTime? readAt;
  final DateTime createdAt;

  const AppNotification({
    required this.id,
    required this.type,
    required this.title,
    required this.body,
    this.entityType,
    this.entityId,
    required this.isRead,
    this.readAt,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, type, title, body, entityType, entityId, isRead, readAt, createdAt];
}
